### ISSUE TITLE
VictoriaMetrics operator: support multiple namespaces `watchNamespace` param

### ISSUE THREAD

**Issue by alexsuraykin** [VictoriaMetrics/operator#557]
Hi everyone,

We need to install two instances of VictoriaMetrics operator on one k8s cluster(it is requirement to separate metrics of infrastructure and cloud from business). And we expected that operator  will manage Vmagent, Vmalert, Vmsingle resources only in self namespace.
But there some errors with VictoriaMetrics operator, it tries to update theis resources from another namespace(second operator).

We also have found and tested env variable `WATCH_NAMESPACE` which described in the issue https://github.com/VictoriaMetrics/operator/pull/329

And it really works, VictoriaMetrics operator starts manage resources only defined namespace, but at the same time it stops managing not only Vmagent, Vmalert, Vmsingle resources, but all VictoriaMetrics's stack resources like vmservicescrapes, vmstaticscrapes, vmpodscrapes, vmrules that can be defined in other namespaces and should be scraped by vmagent/vmalert.

Do you have any assumptions and suggestions which would help us?

P.S. Prometheus operator supports such deployments through next  cmdline flags https://github.com/prometheus-operator/prometheus-operator/blob/main/cmd/operator/main.go#L163-L168.

**Comment by f41gh7** [comment]
Hello, only operator deployment with WATCH_NAMESPACE per namespace.

 Need to think, how to implement it better, for now I don't see any easy way.

**Comment by rishid** [comment]
Just to add a +1 to this issue - it is biting me as well. I used the flags that Alex mentioned in prom-operator, hoping we can get something similar added the VM operator to maintain feature parity.

**Comment by SaschaP95** [comment]
I think this can solve your problem 
https://github.com/VictoriaMetrics/operator/issues/641

**Comment by rauanmayemir** [comment]
Having a `cluster-wide` operator with an option to limit it to only certain namespaces would significantly improve the experience and follow security best practices.

I do want to use a single operator instance for the whole cluster, but I also don't want to include VM into the privileged workloads scope. `vm-operator-manager-role` cluster role requests access to `secrets` resource, meaning it can read all secrets and this privilege is unnecessary. (this is not to mention `clusterroles` and `clusterrolebindings`, VM operator must not need to even touch them)

I was planning to create a separate cluster role `vm-operator-manager-role-scoped` that would operate on all sensitive resources like `secrets`, and create role bindings for the namespaces that run vm clusters, something like this:

```
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: vm-operator-manager-scoped-rolebinding
  namespace: monitoring
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: vm-operator-manager-role-scoped
subjects:
- kind: ServiceAccount
  name: vm-operator
  namespace: monitoring-system
```

That way, I would be able to somewhat decrease the privileges of vm-operator. But that attempt failed with:

```
k8s.io/client-go@v12.0.0+incompatible/tools/cache/reflector.go:169: Failed to watch *v1.PartialObjectMetadata: failed to list *v1.PartialObjectMetadata: secrets is forbidden: User "system:serviceaccount:monitoring-system:vm-operator" cannot list resource "secrets" in API group "" at the cluster scope
```

It would be nice to specify multiple namespaces like this:
```
env:
- name: WATCH_NAMESPACE
  value: namespace1,namespace2
```

A good example where it works like that is [RabbitMQ Operator](https://rabbitmq.com/kubernetes/operator/configure-operator-defaults.html).

**Comment by f41gh7** [comment]
Must be fixed at [v0.42.0 release](https://github.com/VictoriaMetrics/operator/releases/tag/v0.42.0)

 Now operator supports comma separated values for `WATCH_NAMESPACE` env variable.

 It doesn't use any cluster wide permissions. And requires a minimum set of permissions.

 Few issues left for now, operator creates only namespaced rules for vmagent. It could be fixed by external serviceAccount.
